lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -862,6 +863,156 @@ using LowerVectorSubIOpToAIEVecSubOp =
 using LowerVectorSubFOpToAIEVecSubOp =
     OneToOneVectorOpToAIEVecOpPattern<arith::SubFOp, aievec::SubOp>;
 
+template <typename SrcOpTy, typename AIEv2ElemOp>
+static void genAddElemAieML(ConversionPatternRewriter &rewriter, Value lval,
+                            Value rval, VectorType srcType, SrcOpTy srcOp) {
+  auto lCastOp = rewriter.create<aievec::CastOp>(srcOp.getLoc(), srcType, lval,
+                                                 /*isResAcc*/ true);
+  auto rCastOp = rewriter.create<aievec::CastOp>(srcOp.getLoc(), srcType, rval,
+                                                 /*isResAcc*/ true);
+  auto elemOp = rewriter.create<AIEv2ElemOp>(
+      srcOp.getLoc(), lCastOp->getResult(0).getType(), lCastOp->getResult(0),
+      rCastOp->getResult(0));
+  rewriter.replaceOpWithNewOp<aievec::CastOp>(
```

Comment:
Changing the return type to 'LogicalResult' and returning success() allows you to shorten the code in the caller, 
i.e.
`if (blah) {
  genAddElemAieML(..);
  return success();
}`
->
`if (blah)
  return genAddElemAieML(..);
`